Content.Server/Zombies/ZombieSystem.cs

```diff
@@ -176,5 +180,18 @@ public override void Update(float frameTime)
                 DoGroan(zombiecomp.Owner, zombiecomp);
             }
         }
+
+        private void OnZombieCloning(EntityUid uid, ZombieComponent zombiecomp, ref CloningEvent args)
+        {
+            foreach (var (layer, info) in zombiecomp.BeforeZombifiedCustomBaseLayers)
+            {
+                _humanoidSystem.SetBaseLayerColor(args.Target, layer, info.Color);
+                _humanoidSystem.SetBaseLayerId(args.Target, layer, info.ID);
+            }
+            _humanoidSystem.SetSkinColor(args.Target, zombiecomp.BeforeZombifiedSkinColor);
+
+            MetaData(args.Target).EntityName = zombiecomp.BeforeZombifiedEntityName;
+            args.NameHandled = true;
+        }
```

Comment:
make a public unzombify method with this logic then call it here

Reply:
I just did this.